pkg/cache/snapshot_test.go

```diff
@@ -864,3 +864,457 @@ func TestSnapshotAddRemoveWorkload(t *testing.T) {
 		})
 	}
 }
+
+func TestSnapshotAddRemoveWorkloadWithLendingLimit(t *testing.T) {
```

Comment:
This is quite similar to `TestSnapshotAddRemoveWorkload` but since the snapshot result differs with lendingLimit enabled/disabled, I'm ok to make this a separate test.